### ISSUE TITLE
VehicleType offers a Bicycle the API cannot send

### ISSUE THREAD

**Issue by koeylp** [yuthstyle88/108jobs-clean#81]
The generated client type `src/lib/108jobs-client/src/types/VehicleType.ts` declares three variants:

```ts
export type VehicleType =
    | "Bicycle"
    | "Motorcycle"
    | "Car"
```

The backend enum has two. `crates/db/src/enums.rs:380-384` in `api-108jobs`:

```rust
pub enum VehicleType {
  #[default]
  Motorcycle,
  Car,
}
```

No `Bicycle` variant exists, so no response can carry that value.

## Consequence

`src/app/[lang]/admin/manage-riders/page.tsx:20` builds an exhaustive map over the type:

```tsx
const vehicleIconMap: Record<VehicleType, JSX.Element> = {
    Bicycle: <Bike className="w-4 h-4"/>,
    Motorcycle: <Motorbike className="w-4 h-4"/>,
    Car: <Car className="w-4 h-4"/>,
};
```

The `Bicycle` row is unreachable. Harmless today — this drift is in the permissive direction, so it costs a dead branch rather than a runtime failure. It is the opposite of the `licenseNumber` / `licenseExpiryDate` drift fixed alongside this report, where the type claimed fields the API had stopped sending and every read silently produced `undefined`.

It is still a client type that does not describe the server, and anyone writing an exhaustive `switch` on `VehicleType` will write a branch that can never run and may reasonably conclude the platform supports bicycles.

## Fix

Drop `"Bicycle"` from `VehicleType.ts`. Note that doing so turns the `vehicleIconMap` entry above into an excess-property error, so the map has to lose its `Bicycle` row in the same change — which is the point.

Worth checking at the same time whether bicycles were ever intended: if the platform is meant to support them, the fix belongs on the Rust side instead, and this issue is the record of that decision either way.

## How it was found

Reviewing the client-type sync for the rider-application work, comparing every generated type against its Rust source field by field.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
